mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/quota/CassandraQuotaCurrentValueDaoTest.java

```diff
@@ -0,0 +1,103 @@
+
+package org.apache.james.mailbox.cassandra.quota;
+
+import org.apache.james.backends.cassandra.CassandraClusterExtension;
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.blob.cassandra.CassandraBlobModule;
+import org.apache.james.core.Username;
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaCurrentValue;
+import org.apache.james.core.quota.QuotaType;
+import org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils;
+import org.apache.james.mailbox.cassandra.modules.CassandraQuotaModule;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class CassandraQuotaCurrentValueDaoTest {
+
+    private CassandraQuotaCurrentValueDao cassandraQuotaCurrentValueDao;
+
+    @RegisterExtension
+    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraModule.aggregateModules(
+        CassandraBlobModule.MODULE,
+        CassandraQuotaModule.MODULE));
+
+    @BeforeEach
+    private void setup() {
+        cassandraQuotaCurrentValueDao = GuiceUtils.testInjector(cassandraCluster.getCassandraCluster()).getInstance(CassandraQuotaCurrentValueDao.class);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldCreateNewRowSuccessfully() {
+        String username = "james@abc.com";
+        QuotaCurrentValue expected = QuotaCurrentValue.of(QuotaComponent.MAILBOX, Username.of(username), QuotaType.SIZE, 100l);
+        cassandraQuotaCurrentValueDao.increase(expected.getQuotaComponent(), expected.getIdentifier(), expected.getQuotaType(), 100l).block();
+
+        QuotaCurrentValue actual = cassandraQuotaCurrentValueDao.getQuotaCurrentValue(QuotaComponent.MAILBOX, Username.of(username), QuotaType.SIZE).block();
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldCreateNewRowSuccessfullyWhenIncreaseAmountIsZero() {
+        String username = "james@abc.com";
+        QuotaCurrentValue expected = QuotaCurrentValue.of(QuotaComponent.MAILBOX, Username.of(username), QuotaType.SIZE, 0l);
+        cassandraQuotaCurrentValueDao.increase(expected.getQuotaComponent(), expected.getIdentifier(), expected.getQuotaType(), 0l).block();
+
+        QuotaCurrentValue actual = cassandraQuotaCurrentValueDao.getQuotaCurrentValue(QuotaComponent.MAILBOX, Username.of(username), QuotaType.SIZE).block();
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldIncreaseValueSuccessfully() {
+        String username = "james@abc.com";
+        QuotaCurrentValue expected = QuotaCurrentValue.of(QuotaComponent.MAILBOX, Username.of(username), QuotaType.SIZE, 200l);
+        cassandraQuotaCurrentValueDao.increase(expected.getQuotaComponent(), expected.getIdentifier(), expected.getQuotaType(), 100l).block();
```

Comment:
just curious, what happened if increases with a negative number? 
ex: `-100l`

Reply:
it will decrease instead